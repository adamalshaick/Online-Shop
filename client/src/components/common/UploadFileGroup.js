import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const UploadFileGroup = ({ error, icon, type, name, onChange, value }) => {
  return (
    <div className="input-group mt-3">
      <div className="input-group-prepend">
        <div className=" text-center">
          <i className={`${icon} mb-4 ml-3 mt-3`} />
          <input
            className={classnames("form-control", {
              "is-invalid": error
            })}
            onChange={onChange}
            type={type}
            name={name}
            value={value}
            style={{ marginLeft: "1rem" }}
          />
          {error && <div className="invalid-feedback">{error}</div>}
        </div>
      </div>
    </div>
  );
};

UploadFileGroup.propTypes = {
  error: PropTypes.string,
  icon: PropTypes.string,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

UploadFileGroup.defaultProps = {
  type: "file"
};

export default UploadFileGroup;
