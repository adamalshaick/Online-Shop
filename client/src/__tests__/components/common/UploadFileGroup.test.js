import React from "react";
import { shallow } from "enzyme";
import UploadFileGroup from "../../../components/common/UploadFileGroup";
import renderer from "react-test-renderer";

const Props = {
  value: "new value",
  name: "new name",
  onChange: e => {
    e.target.name = e.target.value;
  }
};

describe("Render UploadFileGroup", () => {
  it("render upload file group component", () => {
    const UploadFileComponent = renderer.create(<UploadFileGroup />).toJSON();
    expect(UploadFileComponent).toMatchSnapshot();
  });
});

// beforeEach(() => {
//   wrapped = shallow(<TextFieldGroup {...Props} />);
// });

// it("can type", () => {
//   wrapped.find("input").simulate("change", {
//     target: { value: "new input" }
//   });
//   wrapped.update();
//   expect(wrapped.find("input").prop("value")).toEqual("new value");
// });
