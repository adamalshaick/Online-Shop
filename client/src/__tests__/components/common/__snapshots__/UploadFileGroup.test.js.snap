// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadFileGroup render upload file group component 1`] = `
<div
  className="input-group mt-3"
>
  <div
    className="input-group-prepend"
  >
    <div
      className="card text-center"
    >
      <i />
      <input
        className="form-control form-control-lg"
        style={
          Object {
            "margin": "auto",
          }
        }
        type="file"
      />
    </div>
  </div>
</div>
`;
